packages/sync/src/PersistentStore.ts

```diff
@@ -3,9 +3,9 @@
  * Interface for underlying storage solutions
  */
 export interface PersistentStore<T> {
-  getItem: (key: string) => Promise<T> | T;
-  setItem: (key: string, data: T) => Promise<void> | void;
-  removeItem: (key: string) => Promise<void> | void;
+  getItem: (key: string) => Promise<T>;
+  setItem: (key: string, data: T) => any;
+  removeItem: (key: string) => Promise<void>;
 }
 
 export type PersistedData = string | null | object;
```

Comment:
@StephenCoady We need to aling Storage for data to be string and need to be stringified or object.